cli/azd/pkg/project/service_manager.go

```diff
@@ -258,14 +258,21 @@ func (sm *serviceManager) Package(
 	return async.RunTaskWithProgress(func(task *async.TaskContextWithProgress[*ServicePackageResult, ServiceProgress]) {
 		cachedResult, ok := sm.getOperationResult(ctx, serviceConfig, string(ServiceEventPackage))
 		if ok && cachedResult != nil {
-			task.SetResult(cachedResult.(*ServicePackageResult))
-			return
+			// Static web apps should always be repackaged (enables the `up` flow to work)
+			if serviceConfig.Host != StaticWebAppTarget {
+				cachedResult.(*ServicePackageResult).FromCached = true
+				task.SetResult(cachedResult.(*ServicePackageResult))
+				return
+			}
```

Comment:
We shouldn't be introducing special case scenarios for specific app targets here.  Anything special that a service target needs should happen within the service targets code.